spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+
+  describe 'GET #new' do
+    context 'when user is logged in' do
+
+      let(:user) { create(:user) }
+
+      before(:each) do
+        allow(controller).to receive(:current_user).and_return(user)
+      end
+
+      it 'responds with a 200 OK HTTP response' do
+        get :new
+        expect(response.status).to eq(200)
+      end
+
+
+      it 'renders the new template' do
+        get :new
+        expect(response).to render_template "new"
+      end
+    end
+
+    context 'when user is not logged in' do
+      it 'responds with a 302' do
+        get :new
+        expect(response.status).to eq(302)
+      end
+
+      it 'redirects to the login page' do
+        get :new
+        expect(response).to redirect_to login_path
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with correct params' do
+
+      let(:article) { create(:article) }
+
+      let(:params) { {article: {title: article.title,
+                                content: article.content}} }
```

Comment:
I am setting the variable `article` to equal the factory article, I'm then placing the article values title and content into a hash format in `params` as is needed to pass through the controller.

I would like to use the article factory as it has a foreign key to user which we need to log in and access the page. 

This was the difficulty in testing I working with Shane and James to resolve, if you have a better solution it would be great to meet and discuss it!